modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java

```diff
@@ -634,47 +635,80 @@ public ByteCountingDataInput(InputStream in) {
             this.in = in;
         }
 
+        /** Read bytes from the input stream. */
+        public byte[] read(int len) throws ClientConnectionException {
+            byte[] bytes = new byte[len];
+
+            read(bytes, len, 0);
+
+            return bytes;
+        }
+
+        /** Read bytes from the input stream. */
+        public byte[] readNonBlocking(int len) {
```

Comment:
Let's rename to `readSpinWait` to better reflect the logic. `nonBlocking` is misleading - the method can block the thread.